src/CodeExamples.sln

```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.CalculateCapacity", "HashMap.Test\HashMapNamespace.CalculateCapacity\HashMapNamespace.CalculateCapacity.csproj", "{52CF5877-0FA0-482F-A9E8-137C1FF052EC}"
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ProvideCapacity", "HashMap.Test\HashMapNamespace.ProvideCapacity\HashMapNamespace.ProvideCapacity.csproj", "{6B3046FE-22DE-43B3-9335-83883D159BA1}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Add", "HashMap.Test\HashMapNamespace.Add\HashMapNamespace.Add.csproj", "{8513B597-4E9C-44CD-809F-290F446E21E4}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.AddOrUpdate", "HashMap.Test\HashMapNamespace.AddOrUpdate\HashMapNamespace.AddOrUpdate.csproj", "{DD451542-3BE6-47B2-8D29-B614A0EE2F6B}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Remove", "HashMap.Test\HashMapNamespace.Remove\HashMapNamespace.Remove.csproj", "{2FB8A7E4-D031-466B-8834-DA4DA8265808}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ContainsKey", "HashMap.Test\HashMapNamespace.ContainsKey\HashMapNamespace.ContainsKey.csproj", "{49821606-F17E-4E4A-A1D0-5F5D052120F9}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.TryGetValue", "HashMap.Test\HashMapNamespace.TryGetValue\HashMapNamespace.TryGetValue.csproj", "{E252454D-7F25-45DC-9CF1-C6DB354EB67D}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Clear", "HashMap.Test\HashMapNamespace.Clear\HashMapNamespace.Clear.csproj", "{C54A2E51-5AAA-42BE-BF3D-C5BF7FD9C557}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.GetEnumerator", "HashMap.Test\HashMapNamespace.GetEnumerator\HashMapNamespace.GetEnumerator.csproj", "{CB575984-6982-4487-8944-B4751E0970F0}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.HashMap64Node", "HashMap.Test\HashMapNamespace.HashMap64Node\HashMapNamespace.HashMap64Node.csproj", "{1ABBB916-2C9C-4558-95EE-EDDCEBB4A8A1}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ToString", "HashMap.Test\HashMapNamespace.ToString\HashMapNamespace.ToString.csproj", "{393CAC49-92C5-44D5-9019-96179864A0FF}"
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RealtimePlugin.Run", "RealtimePlugin.Test\RealtimePlugin.Run\RealtimePlugin.Run.csproj", "{75DCAF84-C05A-4CFD-A0EF-A03426D009D4}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleClassLibrary.ComposePlugins", "SimpleClassLibrary.Test\SimpleClassLibrary.ComposePlugins\SimpleClassLibrary.ComposePlugins.csproj", "{EF2FF9B2-CE63-4261-B1B3-931D8491428E}"
```

Comment:
### 2/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj (in 417 ms).
  Restored /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj (in 456 ms).
  1 of 3 projects are up-to-date for restore.
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
/workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs(12,55): warning CS8618: Non-nullable field 'plugins' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(13,55): warning CS8618: Non-nullable field 'plugins' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(15,17): warning CS8618: Non-nullable property 'PluginLocation' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(16,20): warning CS8618: Non-nullable property 'Composer' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
  SimpleClassLibrary -> /workspaces/CodeExamples/src/SimpleClassLibrary/bin/Debug/net6.0/SimpleClassLibrary.dll
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(116,62): warning CS8600: Converting null literal or possible null value to non-nullable type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(116,62): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(124,80): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(134,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
  SimpleClassLibrary.ComposePlugins -> /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/bin/Debug/net7.0/SimpleClassLibrary.ComposePlugins.dll
Test run for /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/bin/Debug/net7.0/SimpleClassLibrary.ComposePlugins.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed TestCompositionOfExistingPlugins [127 ms]
  Error Message:
   System.IO.DirectoryNotFoundException : Could not find a part of the path '/path/to/valid/plugin/directory'.
  Stack Trace:
     at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path, String searchPattern, EnumerationOptions enumerationOptions)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.GetFiles()
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.Initialize(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path)
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.TestCompositionOfExistingPlugins() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 132
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
  Failed TestNullPluginLoader [47 ms]
  Error Message:
     Expected: <System.ArgumentNullException>
  But was:  <System.NullReferenceException: Object reference not set to an instance of an object.
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.<>c__DisplayClass5_0.<TestNullPluginLoader>b__0() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 124
   at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)>

  Stack Trace:
     at SimpleClassLibrary.Test.ComposePluginsTests.TestNullPluginLoader() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 124

  Failed TestSuccessfulPluginComposition [1 ms]
  Error Message:
   System.IO.DirectoryNotFoundException : Could not find a part of the path '/path/to/valid/plugin/directory'.
  Stack Trace:
     at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path, String searchPattern, EnumerationOptions enumerationOptions)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.GetFiles()
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.Initialize(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path)
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.TestSuccessfulPluginComposition() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 99
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)

Failed!  - Failed:     3, Passed:     2, Skipped:     0, Total:     5, Duration: 198 ms - SimpleClassLibrary.ComposePlugins.dll (net7.0)
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins (roost-1709036480) $ 
```